src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers.Common/SyntaxNodeExtensions.cs

```diff
@@ -0,0 +1,212 @@
+        {
+            if (node is null)
+            {
+                return false;
+            }
+
+            return node.IsKind(CS.SyntaxKind.BaseList)
+            || node.IsKind(CS.SyntaxKind.SimpleBaseType)
+            || node.IsKind(VB.SyntaxKind.InheritsStatement);
+        }
+
+        /// <summary>
+        /// Finds the fully qualified name syntax or member access expression syntax (if any)
+        /// that a node is part of and returns that larger, fully qualified syntax node.
+        /// </summary>
+        /// <param name="node">
+        /// The identifier syntax node to find a fully qualified ancestor for. This should be of type
+        /// Microsoft.CodeAnalysis.CSharp.Syntax.NameSyntax or Microsoft.CodeAnalysis.VisualBasic.Syntax.NameSyntax.
+        /// </param>
+        /// <returns>
+        /// Returns the qualified name syntax or member access expression syntax containing the provided
+        /// name syntax. If the node is not part of a larger qualified name, the input node will be returned.
+        /// </returns>
+        public static SyntaxNode GetQualifiedName(this SyntaxNode node)
+        {
+            if (node is null)
+            {
+                throw new ArgumentNullException(nameof(node));
+            }
+
+            // If the node is part of a qualified name, we want to get the full qualified name
+            while (node.Parent is CSSyntax.NameSyntax || node.Parent is VBSyntax.NameSyntax)
+            {
+                node = node.Parent;
+            }
+
+            // If the node is part of a member access expression (a static member access, for example), then the
+            // qualified name will be a member access expression rather than a name syntax.
+            if ((node.Parent is CSSyntax.MemberAccessExpressionSyntax csMAE && csMAE.Name.IsEquivalentTo(node))
+                || (node.Parent is VBSyntax.MemberAccessExpressionSyntax vbMAE && vbMAE.Name.IsEquivalentTo(node)))
+            {
+                node = node.Parent;
+            }
+
+            return node;
+        }
+
+        /// <summary>
+        /// Gets a method declared in a given syntax node.
+        /// </summary>
+        /// <typeparam name="T">The type of method declaration syntax node to search for.</typeparam>
+        /// <param name="node">The syntax node to find the method declaration in.</param>
+        /// <param name="methodName">The name of the method to return.</param>
+        /// <param name="requiredParameterTypes">An optional list of parameter types that the method must accept.</param>
+        /// <returns>The first method declaration in the syntax node with the given name and parameter types, or null if no such method declaration exists.</returns>
+        public static T? GetMethodDeclaration<T>(this SyntaxNode node, string methodName, params string[] requiredParameterTypes)
+            where T : SyntaxNode =>
+            node?.DescendantNodes()
```

Comment:
i prefer not to use expression syntax for scenarios where a break point could be set on multiple lines. Specifically due to error logs that can surface exception by line number. All of this, from a compiler perspective is one line isn't it?

Reply:
Could we make this language agnostic by dropping the generic type?

Reply:
Am trying to remember, do you care about Node expansion here?

Reply:
The problem with dropping the generic type is that it returns that type. We could drop it and make it language agnostic, but would have to return `SyntaxNode` then. I thought it might be better to just get the type of method declaration syntax that's expected (C# or VB) and only find/return nodes of that type.

And yes, it's one line. I'll split it up into more, though, to make it more debuggable.

Reply:
And as far as expanding nodes, I'm just expanding an existing helper method to work with both C# and VB so I'm leaving it as unchanged functionally as possible.